rbac/internal/views.py

```diff
@@ -659,6 +660,89 @@ def list_or_delete_bindings_for_role(request, role_uuid):
         return HttpResponse(json.dumps(info), status=200)
 
 
+def clean_binding_mapping(request, binding_id):
+    """Clean bindingmapping for a role, delete not associated role anymore.
+
+    POST /_private/api/utils/bindings/<binding_id>/clean
+    Params:
+        users: comma separated user ids
+        groups: comma seprated group uuids
+    """
+    if not destructive_ok("api"):
+        return HttpResponse("Destructive operations disallowed.", status=400)
+    if request.method != "POST":
+        return HttpResponse('Invalid method, only "POST" is allowed.', status=405)
+    users = request.GET.get("users")
+    groups = request.GET.get("groups")
+    if not users and not groups:
+        return HttpResponse(
+            'Invalid request, must supply the "users" or "groups" in body.',
+            status=400,
+        )
+
+    replicator = OutboxReplicator()
+    with transaction.atomic():
+        mapping = (
+            BindingMapping.objects.select_for_update()
+            .filter(
+                id=binding_id,
+            )
+            .get()
+        )
+        if users:
+            users = users.split(",")
+            relations_to_remove = []
+            # Check if the user should be removed
+            assert (
+                CrossAccountRequest.objects.filter(user_id__in=users)
+                .filter(roles__id=mapping.role.id)
+                .filter(status="approved")
+                .exists()
+                is False
+            )
+            for user in users:
+                removal = mapping.remove_user_from_bindings(user)
+                if removal is not None:
+                    relations_to_remove.append(removal)
+            # After the migration, if it is still using old format
+            # It must be empty, or there is an issue
+            assert len(mapping.mappings["users"]) == 0
+            mapping.mappings["users"] = {}
+            replicator.replicate(
+                ReplicationEvent(
+                    event_type=ReplicationEventType.EXPIRE_CROSS_ACCOUNT_REQUEST,
+                    info={
+                        "users": users,
+                    },
+                    partition_key=PartitionKey.byEnvironment(),
+                    remove=relations_to_remove,
+                    add=[],
+                ),
+            )
+        if groups:
+            groups = groups.split(", ")
+            relations_to_remove = []
+            assert mapping.role.system is True
+            assert Group.objects.filter(uuid__in=groups).exists() is False
+            for group in groups:
+                removal = mapping.remove_group_from_bindings(group)
+                if removal is not None:
+                    relations_to_remove.append(removal)
+            replicator.replicate(
+                ReplicationEvent(
+                    event_type=ReplicationEventType.MIGRATE_TENANT_GROUPS,
+                    info={
+                        "groups": users,
```

Comment:
I guess groups should be here instead of users ?

Reply:
Good catch, updated